core/source/org/libreoffice/ide/eclipse/core/internal/model/UnoidlProject.java

```diff
@@ -131,6 +150,16 @@ public class UnoidlProject implements IUnoidlProject, IProjectNature {
      */
     private static final String CONFIG_FILE = ".unoproject"; //$NON-NLS-1$
 
+    /**
+     * The name of the package.components file.
+     */
+    private static final String COMPONENTS_FILE = "package.components"; //$NON-NLS-1$
```

Comment:
it should be .component (without s)

Reply:
Not really necessarily. Besides all the examples under [libreoffice/odk/examples/DevelopersGuide/Components](https://github.com/LibreOffice/core/tree/master/odk/examples/DevelopersGuide/Components) show it.

As a good example the file: [JavaComponents.components](https://github.com/LibreOffice/core/blob/master/odk/examples/DevelopersGuide/Components/JavaComponent/JavaComponent.components)

It appears that there can be two types of components:
- Simple components whose root of the XML document is the `<component>` element.
- Multiple components whose root of the XML document is the `<components>` element.

I think the file extension follows this logic? At least that's what I thought.